web/src/components/RawDataEditor/RawDataEditor.tsx

```diff
@@ -38,13 +38,17 @@ enum codeMirrorModeList {
   Yaml = 'Yaml',
 }
 
-const RawDataEditor: React.FC<Props> = ({ visible, readonly = true, type, data = {}, onClose, onSubmit = () => { } }) => {
+const RawDataEditor: React.FC<Props> = ({ visible, readonly = true, type, data = {}, onClose = () => { }, onSubmit = () => { } }) => {
   const ref = useRef<any>(null);
   const { formatMessage } = useIntl();
   const [codeMirrorMode, setCodeMirrorMode] = useState<PluginComponent.CodeMirrorMode>(
     codeMirrorModeList.Json,
   );
 
+  useEffect(() => {
+    setCodeMirrorMode(codeMirrorModeList.Json);
```

Comment:
only visible, right?

Reply:
Not really, setting the Drawer to Json when closing it will ensure that the state before visible is Json.

Reply:
why set code mirror mode in visible?